internal/controller/spinapp_controller.go

```diff
@@ -381,14 +391,7 @@ func (r *SpinAppReconciler) deleteDeployment(ctx context.Context, app *spinv1alp
 // constructDeployment builds an appsv1.Deployment based on the configuration of a SpinApp.
 func constructDeployment(ctx context.Context, app *spinv1alpha1.SpinApp, config *spinv1alpha1.ExecutorDeploymentConfig,
 	generatedRuntimeConfigSecretName, caSecretName string, scheme *runtime.Scheme) (*appsv1.Deployment, error) {
-	// TODO: Once we land admission webhooks write some validation to make
-	// replicas and enableAutoscaling mutually exclusive.
-	var replicas *int32
-	if app.Spec.EnableAutoscaling {
-		replicas = nil
-	} else {
-		replicas = ptr(app.Spec.Replicas)
-	}
+	replicas := app.Spec.Replicas
```

Comment:
related to the above comment RE: `min/maxReplicas`, shouldn't this be `nil` if either of those are set?